app/javascript/react/components/Map/Map.tsx

```diff
@@ -1,19 +1,92 @@
-import React from "react";
-import { useSelector } from "react-redux";
-
-import { Map as GoogleMap } from "@vis.gl/react-google-maps";
+import React, { useEffect, useState, useCallback } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Map as GoogleMap, MapEvent } from "@vis.gl/react-google-maps";
 
+import mapStyles from "./mapStyles";
 import { DEFAULT_MAP_CENTER, DEFAULT_ZOOM } from "../../const/coordinates";
 import { RootState } from "../../store";
+import { fetchFixedSessions } from "../../store/fixedSessionsSlice";
 import { containerStyle } from "./Map.style";
-import mapStyles from "./mapStyles";
 import { Markers } from "./Markers/Markers";
-import trees from "./trees";
+import { selectSessionsData } from "../../store/fixedSessionsSelectors";
+import { Session } from "./Markers/SessionType";
+import { useAppDispatch } from "../../store/hooks";
 
 const Map = () => {
+  const dispatch = useAppDispatch();
+
+  const timeFrom = "1685318400";
+  const timeTo = "1717027199";
+  const tags = "";
+  const usernames = "";
+  const limit = 100;
+  const offset = 0;
+  const sensor_name = "government-pm2.5";
+  const measurement_type = "Particulate Matter";
+  const unit_symbol = "µg/m³";
+
+  const [mapBounds, setMapBounds] = useState({
+    north: 47.886881016621686,
+    south: 24.507143507735677,
+    east: -59.018555062500006,
+    west: -132.4072269375,
+  });
```

Comment:
You can move these coordinates to a separate file - in the project we have `coordinates.ts `